### DISCUSSION TITLE
Index best practice. Is there a limit? Will it affect performance? Whats the Maximum Read/s Write/s per DB?

### DISCUSSION THREAD

**Discussion by jericopulvera** [planetscale/discussion#180] [FAQs] [answered]
From what I understand when you index a column it would take up more disk space and slow down write speed.

In the Planetscale case, how does it affect the performance?

What's the maximum Read/s Write/s per DB?

**Comment by jgreet** [comment, marked as answer]
In general, just like any other relational database, adding an additional index to a table will make INSERT statements slower but make previously unindexed SELECT statements that can use the new index faster. You don't want to add unused or duplicate indexes, but if you have a SELECT query that needs to be faster, you likely need an index.

From https://use-the-index-luke.com/sql/dml/update

> Considering insert statements only, it would be best to avoid indexes entirely—this yields by far the best insert performance. However tables without indexes are rather unrealistic in real world applications. You usually want to retrieve the stored data again so that you need indexes to improve query speed. Even write-only log tables often have a primary key and a respective index.

With PlanetScale there's the additional wrinkle of [usage based billing](https://docs.planetscale.com/concepts/billing#understanding-rows-read). Those row reads can add up quickly if a frequent query triggers a full table scan on a large table. 

Max reads and writes are going to depend on the hardware provisioned for your database cluster. Our own application reads more than 100k rows per second but that wouldn't be possible on a free tier database. If there's a specific workload you're worried about moving over it might be worth reaching out to our [sales engineers](https://planetscale.com/contact).

**Reply by jericopulvera** [reply]
What are the maximum read/s and write/s in the free tier and scaler plan?

**Reply by KaraboGerald** [reply]
I'm on the free tier, getting around 13 rows/s (writes) which is the max I've seen concurrently over an hour of writing stock market data, but the reads are higher.
